src/node/mod.rs

```diff
@@ -517,31 +519,24 @@ impl<N: NodeData<J>, J: Jet> Node<N, J> {
         &self.data
     }
 
-    /// Generic conversion function from one type of node to another.
+    /// Generic conversion function from one type of node to another, with the
+    /// ability to prune during the conversion.
     ///
-    /// Parameterized over what kind of sharing to use when iterating over
-    /// the original DAG.
+    /// Parameterized over what kind of sharing to use when iterating over the
+    /// DAG, and what conversion logic to use.
     ///
-    /// Generally users won't need to use this function directly; it is used
-    /// internally to convert a `ConstructNode` to a `CommitNode` by finalizing
-    /// types, and to convert a `CommitNode` to a `RedeemNOde` by attaching
-    /// witnesses.
-    pub fn convert<S, FD, FW, M, E>(
-        &self,
-        mut convert_data: FD,
-        mut convert_witness: FW,
-    ) -> Result<Arc<Node<M, J>>, E>
+    /// See the documentation for [`Converter`] for details.
+    pub fn convert<S, M, C>(&self, converter: &mut C) -> Result<Arc<Node<M, J>>, C::Error>
     where
-        S: for<'a> SharingTracker<&'a Self> + Default,
+        S: for<'a> SharingTracker<SwapChildren<&'a Self>> + Default,
         M: NodeData<J>,
-        FD: FnMut(
-            &PostOrderIterItem<&Self>,
-            Inner<&Arc<Node<M, J>>, J, M::Witness>,
-        ) -> Result<M::CachedData, E>,
-        FW: FnMut(&PostOrderIterItem<&Self>, &N::Witness) -> Result<M::Witness, E>,
+        C: Converter<N, M, J>,
     {
         let mut converted: Vec<Arc<Node<M, J>>> = vec![];
-        for data in self.post_order_iter::<S>() {
+        for data in self.rtl_post_order_iter::<S>() {
```

Comment:
Lol, oops, I didn't mean to use the `rtl_` variant here. I'll adjust the next PR to change it.